packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -597,6 +611,174 @@ export class DotNetGenerator extends Generator {
     }
   }
 
+  /**
+   * Emints type checks for values passed for type union parameters.
+   *
+   * @param parameters the list of parameters received by the function.
+   * @param noMangle   use parameter names as-is (useful for setters, for example) instead of mangling them.
+   */
+  private emitUnionParameterValdation(
+    parameters?: readonly spec.Parameter[],
+    { noMangle = false }: { noMangle?: boolean } = {},
+  ): void {
+    const unionParameters = parameters?.filter(({ type }) =>
+      containsUnionType(type),
+    );
+    if (unionParameters == null || unionParameters.length === 0) {
+      return;
+    }
+    this.code.indent('#if DEBUG');
+    for (const param of unionParameters) {
+      const name = noMangle
+        ? param.name
+        : this.nameutils.convertParameterName(param.name);
+
+      if (param.optional) {
+        this.code.openBlock(`if (${name} != null)`);
+      }
+
+      validate.call(
```

Comment:
I need to ensure the correct value for `this` is provided, and using `.call` is the only way (it otherwise fails to satisfy type checker requirements).